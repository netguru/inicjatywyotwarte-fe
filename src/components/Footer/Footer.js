import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

const MainContainer = styled.div`
  border-top: 1px solid ${props => props.theme.modalColorVeryLight};
  height: ${props => props.theme.footerHeight};
  width: 100%;
  margin: ${props => props.theme.footerHeight} auto 0 auto;

  @media (max-width: 630px) {
    height: calc(2 * ${props => props.theme.footerHeight});
  }
`

const ContentWrapper = styled.div`
  width: ${props => props.theme.resourceTilesWidth};
  margin: 0 auto;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: ${props => props.theme.fontColorUpvotesCount};
  font-size: 0.8rem;
  font-weight: 300;
  
  @media (max-width: 1000px) {
    width: ${props => props.theme.mobileResourceTilesWidth};
  }

  @media (max-width: 630px) {
    flex-direction: column;
  }
`

const RightContent = styled.div`
  display: flex;
  justify-content: space-between;
  height: 100%;
`

const LeftContent = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
`

const LinkWrapper = styled.div`
  margin-left: 50px;
  height: 100%;

  @media (max-width: 630px) {
    margin: 0 1.5rem;
  }
`

const StyledLink = styled(Link)`
  color: inherit;
  text-decoration: none;
  text-align: center;
  cursor: pointer;
  display: flex;
  align-items: center;
  height: 100%;
  transition: ${props => props.theme.buttonTransition};
  white-space: nowrap;

  &:hover {
    color: ${props => props.theme.linkHoverLightColor};
  }
`
const Footer = () => {
  return (
    <MainContainer>
      <ContentWrapper>
        <LeftContent>Inicjatywy Otwarte &copy; 2020</LeftContent>
        <RightContent>
          <LinkWrapper>
            <StyledLink to='/polityka-prywatnosci'>
              Polityka <br />
              prywatno≈õci
            </StyledLink>
          </LinkWrapper>
          <LinkWrapper>
            <StyledLink to='/regulamin-strony'>
              Regulamin <br />
              strony
            </StyledLink>
          </LinkWrapper>
          <LinkWrapper>
            <StyledLink to='/pomoc'>Pomoc</StyledLink>
          </LinkWrapper>
          <LinkWrapper>
            <StyledLink to='/o-nas'>O nas</StyledLink>
          </LinkWrapper>
        </RightContent>
      </ContentWrapper>
    </MainContainer>
  )
}

export default React.memo(Footer)
