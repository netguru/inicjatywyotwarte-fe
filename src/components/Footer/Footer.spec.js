import React from 'react'
import Footer from './Footer'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Footer', () => {
  test('renders footer', () => {
    const wrapper = shallow(<Footer/>);
    expect(wrapper.find('FooterMainContainer').length).toEqual(1);
  });

  test('renders proper links', () => {
    const wrapper = shallow(<Footer/>);

    expect(wrapper.find('[to="/polityka-prywatnosci"]').length).toEqual(1);
    expect(wrapper.find('[to="/regulamin-strony"]').length).toEqual(1);
    expect(wrapper.find('[to="/pomoc"]').length).toEqual(1);
    expect(wrapper.find('[to="/o-nas"]').length).toEqual(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<Footer/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


