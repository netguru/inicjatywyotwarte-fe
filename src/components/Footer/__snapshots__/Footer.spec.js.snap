// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer match snapshot 1`] = `
<FooterMainContainer>
  <styled.div>
    <styled.div>
      Inicjatywy Otwarte © 2020
    </styled.div>
    <styled.div>
      <styled.div>
        <Styled(Link)
          to="/polityka-prywatnosci"
        >
          Polityka 
          <br />
          prywatności
        </Styled(Link)>
      </styled.div>
      <styled.div>
        <Styled(Link)
          to="/regulamin-strony"
        >
          Regulamin 
          <br />
          strony
        </Styled(Link)>
      </styled.div>
      <styled.div>
        <Styled(Link)
          to="/pomoc"
        >
          Pomoc
        </Styled(Link)>
      </styled.div>
      <styled.div>
        <Styled(Link)
          to="/o-nas"
        >
          O nas
        </Styled(Link)>
      </styled.div>
    </styled.div>
  </styled.div>
</FooterMainContainer>
`;
